### PR TITLE
Initial Android support attempt

### PR DESCRIPTION
**Pull request by io12**
I've been trying to port Box86 to Android to run the x86 Android version of Wine. I've gotten pretty far and it mostly works now, but wineboot fails with an internal error. It would be great if you could look over this PR and see if there are any obvious problems with it.

**CMake flags**:
```
cmake .. -DARM_DYNAREC=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DANDROID=1 -DNOLOADADDR=1 -DANDROID_ABI=armeabi-v7a -DCMAKE_TOOLCHAIN_FILE=/opt/android-ndk/build/cmake/android.toolchain.cmake -DANDROID_NATIVE_API_LEVEL=28 -DHAVE_TRACE=1
```

**Wine version used**: https://dl.winehq.org/wine-builds/android/wine-6.5-x86.apk

**Normal log**:
<details>

```
~/box86/build $ BOX86_DYNAREC=0 PATH=$PATH:~/wine/assets/x86/bin ./box86t wine wineboot
Box86 with trace with Dynarec v0.2.1 9d090bc5 built on Apr 12 2021 11:01:19
Box86 with trace with Dynarec v0.2.1 9d090bc5 built on Apr 12 2021 11:01:19
Box86 with trace with Dynarec v0.2.1 9d090bc5 built on Apr 12 2021 11:01:19
0024:err:environ:run_wineboot failed to start wineboot c00000e5
Warning, accessing segment NULL
11135|SIGSEGV @0xad3d85ac (???(/data/data/com.termux/files/home/box86/build/box86t/0xad3d85ac)) (x86pc=0xf5059cdb//data/data/com.termux/files/home/wine/assets/x86/bin/../lib/wine/ntdll.so:"???", esp=0xf491df80, stack=0xf411f000:0xf491f000 own=0x0 fp=0xf491e284), for accessing 0x1f8 (code=1/prot=0), db=0x0(0x0:0x0/0x0:0x0/???:clean, hash:0/0)
11135|Double SIGSEGV (code=1, pc=0xad3d85ac, addr=0x1f8)!
```

</details>

**Log with `WINEDEBUG=+all`**:
[log1.txt](https://github.com/ptitSeb/box86/files/6296196/log1.txt)

**Log with `WINEDEBUG=+all BOX86_LOG=2`**:
[log2.txt](https://github.com/ptitSeb/box86/files/6296197/log2.txt)

### CODE REVIEW

File: src/wrapped/gnulib_qsort.c [hunk 1]
```diff
@@ -0,0 +1,252 @@
+/* Copyright (C) 1991-2021 Free Software Foundation, Inc.
```
Comment (ptitSeb) [inline src/wrapped/gnulib_qsort.c:1]: I cannot accept this file. GPL is not compatible with MIT licence used in box86.

**Review by ptitSeb — COMMENTED** [review]
I have seen nothing obvious in the trace not in the code for now. I did a few comment, check them please.

File: src/library_list.h [hunk 2]
```diff
@@ -138,7 +138,7 @@ GO("libtiff.so.5", libtiff)
 
 #ifdef ANDROID
 GO("libc.so", libc)
-GO("libm.so", libc)
+GO("libm.so", libm)
```
Comment (ptitSeb) [inline src/library_list.h:141]: Ah yes, that will help things!

File: CMakeLists.txt [hunk 3]
```diff
@@ -10,6 +10,7 @@ option(RK3399 "Set to ON if targeting an Rockchip RK3399 based device" ${RK3399}
 option(GAMESHELL "Set to ON if targeting a GameShell device" ${GAMESHELL})
 option(ODROID "Set to ON if targeting an ODroid device" ${ODROID})
 option(POWER9 "Set to ON if targeting a POWER9 processor" ${POWER9})
+option(ANDROID "Set to ON if targeting an Android device" ${ODROID})
```
[the remark this answers is not in the document]
Comment (io12) [reply CMakeLists.txt]: Fixed, thanks!

File: src/wrapped/gnulib_qsort.c (hunk 1 above)
Comment (io12) [reply src/wrapped/gnulib_qsort.c]: Okay, I wrote a custom `qsort_r()`.

**Comment by io12** [comment]
It looks like wine 6.3 works with Dynarec off! (I only tested cmd.exe because I don't think the downloadable Android wine builds use X11) I'm not sure why newer versions of wine fail to start wineboot or why Dynarec causes SIGILL, but it's progress.

**Comment by ptitSeb** [comment]
Ok, thanks. Is it ready to merge or do you still have some stuff you want to do before?

**Comment by io12** [comment]
I think it's ready. I squashed the commits so the gnulib file isn't in the history.

**Pull request merged**